addons/web/static/src/views/list/list_controller.js

```diff
@@ -107,11 +107,8 @@ export class ListController extends Component {
             rootRef: this.rootRef,
             beforeLeave: async () => {
                 const list = this.model.root;
-                const editedRecord = list.editedRecord;
-                if (editedRecord) {
-                    if (!(await list.unselectRecord(true))) {
-                        return false;
-                    }
+                if (!(await list.leaveEditMode())) {
+                    return false;
```

Comment:
```suggestion
                return list.leaveEditMode();
```

Reply:
We do not need to wait for the saving result ?

Reply:
it's equivalent, in both cases you return a promise that resolves in a boolean (for the `false` case at least)

Reply:
Right, updated :)